.github/workflows/build.yml

```diff
@@ -112,6 +112,67 @@ jobs:
             ${{ github.workspace }}/packages/*.tar.gz
             ${{ github.workspace }}/packages/*.sha256
 
+  macos-universal-package:
+    runs-on: macos-11
+    needs: [macos-build, macos-cross-build]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          path: artifacts
+
+      - name: Create universal binary
+        run: |
+          # Create temporal folders
+          mkdir -p ./pack-temp/{osx-x64,osx-arm64,osx}
+
+          # Extract x86_64 and arm64 artifacts
+          tar -zxf ./artifacts/libddwaf-macos-x86_64/libddwaf-*-darwin-x86_64*.tar.gz -C ./pack-temp/osx-x64
+          tar -zxf ./artifacts/libddwaf-macos-arm64/libddwaf-*-darwin-arm64*.tar.gz -C ./pack-temp/osx-arm64
+
+          # Copy one as a base for the universal artifact (headers and cmake files doesn't contains any arch data, so it's safe to use for the universal package)
+          cp -R ./pack-temp/osx-x64/* ./pack-temp/osx
+
+          # Declare names vars
+          x64Name=$(cd ./pack-temp/osx-x64 && ls)
+          arm64Name=$(cd ./pack-temp/osx-arm64 && ls)
+          universalName=$((cd ./pack-temp/osx && ls) | sed 's/x86_64/universal/g')
+
+          # Change the current folder name with the universal name
+          mv ./pack-temp/osx/libddwaf-*-darwin-* ./pack-temp/osx/$universalName
+
+          # Create the universal binary for the shared library: libddwaf.dylib
+          lipo ./pack-temp/osx-x64/$x64Name/lib/libddwaf.dylib ./pack-temp/osx-arm64/$arm64Name/lib/libddwaf.dylib -create -output ./pack-temp/osx/$universalName/lib/libddwaf.dylib
+          # Check the universal binary
+          lipo -archs ./pack-temp/osx/$universalName/lib/libddwaf.dylib | grep -q 'x86_64 arm64' && echo 'archs ok'
+
+          # Create the universal binary for the static library: libddwaf.a.stripped
+          lipo ./pack-temp/osx-x64/$x64Name/lib/libddwaf.a.stripped ./pack-temp/osx-arm64/$arm64Name/lib/libddwaf.a.stripped -create -output ./pack-temp/osx/$universalName/lib/libddwaf.a.stripped
+          # Check the universal binary
+          lipo -archs ./pack-temp/osx/$universalName/lib/libddwaf.a.stripped | grep -q 'x86_64 arm64' && echo 'archs ok'
```

Comment:
Watch out for priority (I can't seem to ever recall if it's `|` before `&&` or the other way around):

```
(lipo -archs ./pack-temp/osx/$universalName/lib/libddwaf.a.stripped | grep -q 'x86_64 arm64') && echo 'archs ok'
# vs
lipo -archs ./pack-temp/osx/$universalName/lib/libddwaf.a.stripped | (grep -q 'x86_64 arm64' && echo 'archs ok')
```

Can't recall either if `set -o pipefail` is set in these shells, which may have an influence.

Also also, subshells vs `set -e`.

(not saying this line is not working as intended, just raising awareness, as it can be easy to have the failure case not actually fail)

Reply:
Good catch, added some parentheses in [9bffa94](https://github.com/DataDog/libddwaf/pull/136/commits/9bffa944740e99be720dcf60d1046b18075f3e42)